src/ImageSharp/Formats/Png/PngEncoderCore.cs

```diff
@@ -738,6 +753,78 @@ private void WriteExifChunk(Stream stream, ImageMetadata meta)
             }
         }
 
+        /// <summary>
+        /// Writes a text chunk to the stream. Can be either a tTXt, iTXt or zTXt chunk,
+        /// depending whether the text contains none latin character or should be compressed.
+        /// </summary>
+        /// <param name="stream">The <see cref="Stream"/> containing image data.</param>
+        /// <param name="meta">The image metadata.</param>
+        private void WriteTextChunks(Stream stream, ImageMetadata meta)
+        {
+            const int MaxLatinCode = 255;
+            foreach (ImageProperty imageProperty in meta.Properties)
+            {
+                bool hasUnicodeCharacters = imageProperty.Value.Any(c => c > MaxLatinCode);
+                if (hasUnicodeCharacters)
+                {
+                    // Write iTXt chunk.
+                    byte[] keywordBytes = this.latinEncoding.GetBytes(imageProperty.Name);
+                    byte[] textBytes = imageProperty.Value.Length > this.compressTextThreshold ? this.GetCompressedTextBytes(Encoding.UTF8.GetBytes(imageProperty.Value)) : Encoding.UTF8.GetBytes(imageProperty.Value);
+
+                    // Note: The optional language tag and the translated keyword will be omitted.
+                    Span<byte> outputBytes = new byte[keywordBytes.Length + textBytes.Length + 5];
```

Comment:
This is why I was concerned about using the `ImageProperty` type. I think these properties should all be specific to Png Metadata only. 

In fact, I think we should dump `ImageProperty`!

Reply:
ok i will create a png specific TextProperty then, with a language tag and a translated keyword.

But i think we can not dump ImageProperty because its used in GifDecoder, too.

Reply:
If you can replace the property in the Gif meta with a specific one as part of this PR, id' be happy. It should only be a minor change, but if not, we can wait and do it separately.

Reply:
i have added a png specific TextDataProperty and the old `ImageProperty` is now the new Gif specific TextProperty. I dont know about any gif textdata (are there any restrictions?), i assume its similar to the PNG textdata and can hold  something like Author, Comment, Description etc.

Reply:
So it looks like we've been doing Gif comment blocks incorrectly. We should be checking for sub blocks within the extension block. 

https://tronche.com/computer-graphics/gif/gif89a.html#comment-extension

Reply:
i have just tried to add a comment to an image with imagemagick and read it with ImageSharp.
I think the decoder handles it correctly:
- Checks first for ExtensionIntroducer
- Then length of the comment
- The actual string
- A final 0 byte to terminate

What i noticed reading the gif specs is, that gif comments does not have a Name for the Propertys like PNG. Maybe we should remove the Name field of the GifTextData then?

Reply:
I'm not sure it is right. The code in the master branch appears to treat each comment sub block as a distinct comment which is incorrect and also check against and arbitrary maximum length.

My interpretation of the [spec ](https://www.w3.org/Graphics/GIF/spec-gif89a.txt) section highlighted below is as follows:

- There can be multiple Comments in a gif.
- Each Comment is made up of multiple sub blocks stored in blocks ranging from 1-255 bytes in length.
- There is no maximum size of a Comment

So by my reckoning we should have a `Comments` property within the Gif metadata which is a `List<string>`.

What do you think? Apologies btw for throwing all this at you!

```
24. Comment Extension.

      a. Description. The Comment Extension contains textual information which
      is not part of the actual graphics in the GIF Data Stream. It is suitable
      for including comments about the graphics, credits, descriptions or any
      other type of non-control and non-graphic data.  The Comment Extension
      may be ignored by the decoder, or it may be saved for later processing;
      under no circumstances should a Comment Extension disrupt or interfere
      with the processing of the Data Stream.

      This block is OPTIONAL; any number of them may appear in the Data Stream.

      b. Required Version.  89a.

      c. Syntax.

      7 6 5 4 3 2 1 0        Field Name                    Type
     +---------------+
  0  |               |       Extension Introducer          Byte
     +---------------+
  1  |               |       Comment Label                 Byte
     +---------------+

     +===============+
     |               |
  N  |               |       Comment Data                  Data Sub-blocks
     |               |
     +===============+

     +---------------+
  0  |               |       Block Terminator              Byte
     +---------------+

            i) Extension Introducer - Identifies the beginning of an extension
            block. This field contains the fixed value 0x21.

            ii) Comment Label - Identifies the block as a Comment Extension.
            This field contains the fixed value 0xFE.

            iii) Comment Data - Sequence of sub-blocks, each of size at most
            255 bytes and at least 1 byte, with the size in a byte preceding
            the data.  The end of the sequence is marked by the Block
            Terminator.

            iv) Block Terminator - This zero-length data block marks the end of
            the Comment Extension.

      d. Extensions and Scope. This block does not have scope. This block
      cannot be modified by any extension.

      e. Recommendations.

            i) Data - This block is intended for humans.  It should contain
            text using the 7-bit ASCII character set. This block should
            not be used to store control information for custom processing.

            ii) Position - This block may appear at any point in the Data
            Stream at which a block can begin; however, it is recommended that
            Comment Extensions do not interfere with Control or Data blocks;
            they should be located at the beginning or at the end of the Data
            Stream to the extent possible.
```

Reply:
I have read the spec again and i think you are right. I believe i can change the gif encoder / decoder to follow the spec.

Reply:
You rock! 🤘

Reply:
ok there is now a first version ready for review for the changes to the gif encoder / decoder for the comments. Let me know what you think.